ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/KtLintKLoggerInitializer.kt

```diff
@@ -12,8 +12,26 @@ public const val KTLINT_UNIT_TEST_DUMP_AST = "KTLINT_UNIT_TEST_DUMP_AST"
 public const val KTLINT_UNIT_TEST_ON_PROPERTY = "ON"
 
 /**
- * Initializes the logger. Optionally the logger can be modified using the [loggerModifier].
+ * Default modifier for the KLogger of new instances of classes calling [initKtLintKLogger]. Classes for which
+ * [initKtLintKLogger] has been called before setting this variable will not be changed. Also note, that this modifier
+ * can only be set once.
+ */
+public lateinit var defaultLoggerModifier: (KLogger) -> Unit
```

Comment:
> Also note, that this modifier can only be set once.
Are there any checks for that? `lateinit` only checks whether the variable has been initialized before it is accessed, no additional checks on its modification are performed.

Reply:
Tnx again. That was a rookie mistake. I have improved it in #1430. I can not prevent the variable from being set more than once as it will break some integration tests in ktlint. I have now succeeded to print a warning (to discourage doing it) whenever it happens.